### ISSUE TITLE
Problem with lemma type printing in the Outline buffer

### ISSUE THREAD

**Issue by KostiaChardonnet** [cpitclaudel/company-coq#195]
Hi, 

If a Lemma type is too long and wrote on multiple lines, the type is cut in the Outline buffer : 

Here is the lemma in my file : 
`Lemma get_app_l {A} (l l':list A)(n:nat) : n < length l ->
  list_get (l++l') n = list_get l n.
Proof.
 revert l.
 induction n; destruct l; simpl; auto with arith; inversion 1.
Qed.
`

Here is the result in the Outline buffer  : 
`Lemma get_app_l {A} (l l':list A)(n:nat) : n < length l ->`

**Comment by cpitclaudel** [comment]
Indeed, this is mostly on purpose.  You can use code folding (C-c C-\) to hide proofs while leaving all types visible.  Does that work better?

**Comment by KostiaChardonnet** [comment]
Hi, sorry for my late answer.

Code folding only works for the Coq buffer, not the Outline.

The problem with code folding on the Coq buffer is that is it not compacted enough if you just want to search for a definition / lemma and jump on it (which is possible with the Outline)

**Comment by cpitclaudel** [comment]
OK, got it.  I edited your original post (all code was on one line, which got me confused).

Can you give a mock-up of what your ideal buffer would look like, though? I'm still not entirely sure what the right intermediate is between lemma statements and full folding.

**Comment by KostiaChardonnet** [comment]
Hi,

So I'm not too sure on how to give you a mock-up but instead of having this  : (current Outline buffer, the whole type of the Lemma is not printed)

![image](https://user-images.githubusercontent.com/20503424/40017208-f890cc32-57b8-11e8-9cff-155f3e4b8bd1.png)

We would have something like this : 

![image](https://user-images.githubusercontent.com/20503424/40016927-2c9fe2e8-57b8-11e8-86b9-33ec31bcccd1.png)

Or even like this : 

![image](https://user-images.githubusercontent.com/20503424/40017152-d3a5d82c-57b8-11e8-9829-80a7138e9939.png)

(Sorry for the really poor editing)
